common/network-common/src/test/java/org/apache/spark/network/ssl/ReloadingX509TrustManagerSuite.java

```diff
@@ -0,0 +1,320 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.network.ssl;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.security.KeyPair;
+import java.security.KeyStore;
+import java.security.cert.X509Certificate;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import static org.apache.spark.network.ssl.SslSampleConfigs.*;
+
+/**
+ *
+ */
+public class ReloadingX509TrustManagerSuite {
+
+  private final Logger logger = LoggerFactory.getLogger(ReloadingX509TrustManagerSuite.class);
+
+  /**
+   * Waits until reload count hits the requested value, sleeping 100ms at a time.
+   * If the maximum number of attempts is hit, throws a RuntimeException
+   * @param tm the trust manager to wait for
+   * @param count The count to wait for
+   * @param attempts The number of attempts to wait for
+   */
+  private void waitForReloadCount(ReloadingX509TrustManager tm, int count, int attempts)
```

Comment:
We are using 1 second for the tests, given the tests will terminate quicker in case the required condition is reached sooner - we should probably bump that to higher duration to minimize flakiness in case the nodes are loaded.

Unfortunately, our values are all over the place in tests - 3, 5, 10, etc .... I would suggest using 5s and ensure we dont need to revisit this unless it is an actual failure and not flakiness.